contracts/Governance.sol

```diff
@@ -0,0 +1,925 @@
+    mapping(bytes32 => mapping(address => bool)) public votersRoles;
+    mapping(address => mapping(bytes32 => mapping(address => uint256)))
+        public delegations;
+    mapping(address => mapping(bytes32 => uint256)) public delegatees;
+    mapping(address => mapping(bytes32 => uint256)) public delegated;
+    mapping(address => mapping(bytes32 => bytes32)) public actionsRoles;
+    mapping(address => mapping(bytes32 => uint256)) public actionsQuorums;
+    mapping(bytes32 => uint256) public proposalThresholds;
+    mapping(bytes32 => address) public rolesChildrenDAO;
+
+    mapping(uint256 => Proposal) private _proposals;
+    mapping(address => uint256) private _nonces;
+
+    event ProposalCreated(
+        uint256 proposalId,
+        address proposer,
+        address[] targets,
+        uint256[] values,
+        bytes32[] roles,
+        bytes[] calldatas,
+        string[] signatures,
+        bytes32[] actions,
+        uint256 startBlock,
+        uint256 endBlock,
+        string description
+    );
+    event ProposalCanceled(uint256 proposalId);
+    event ProposalExecuted(uint256 proposalId);
+    event ProposalQueued(uint256 proposalId, uint256 eta);
+    event TimelockChange(address oldTimelock, address newTimelock);
+    event VoteCast(
+        address indexed voter,
+        uint256 proposalId,
+        uint8 support,
+        string reason
+    );
+
+    /**
+     * @dev Restrict access to governor executing address. Some module might override the _executor function to make
+     * sure this modifier is consistant with the execution model.
+     */
+    modifier onlyGovernance() {
+        require(_msgSender() == _executor(), 'Governor: onlyGovernance');
+        _;
+    }
+
+    modifier roleExists(bytes32 role) {
+        require(_roles[role], 'Governance::roleExists: role does not exist');
+        _;
+    }
+
+    modifier hasRole(bytes32 role, address delegatee) {
+        require(
+            votersRoles[role][delegatee],
+            'Governance::hasRole: delegatee does not have the role'
+        );
+        _;
+    }
+
+    constructor(IERC20Votes token_, TimelockController timelock_) EIP712(name(), version()) {
+        token = token_;
+        _timelock = timelock_;
+    }
+
+    /**
+     * @dev Address through which the governor executes action. Will be overloaded by module that execute actions
+     * through another contract such as a timelock.
+     */
+    function _executor() internal view virtual returns (address) {
+        return address(this);
+    }
+
+    /**
+     * @dev Function to receive ETH that will be handled by the governor (disabled if executor is a third party contract)
+     */
+    receive() external payable virtual {
+        require(_executor() == address(this));
+    }
+
+    function version() public pure virtual returns (string memory) {
+        return '0.0.1';
+    }
+
+    function name() public pure virtual returns (string memory) {
```

Comment:
`BitDAO` is the same of the ERC20 contract, perhaps something else would suit this contract better, `Governance` perhaps?

Reply:
I think this still needs to be discussed, but I can integrate into governance some of the ERC20 comp func.